sympy/core/function.py

```diff
@@ -512,8 +512,7 @@ def _should_evalf(cls, arg):
         m = pure_complex(arg)
         if m is None or not (m[0].is_Float or m[1].is_Float):
```

Comment:
@eagleoflqj You are right. I tried to find an example of this to add a regression test, but could not find one. The result from `pure_complex` is only not `None` if it is of the form `a+b*i`, which `a` and `b` a `Number`. The only two options are a `Float` or a `Rational`, which both seem to work fine. Do you have an example?

Any subclass of `Number` has a field `_prec`, so we could even skip the checks on `m[i].is_Float`.

I updated the PR, but still have the checks on `is_Float` in place.

Reply:
The check `(m[0].is_Float or m[1].is_Float)1` needs to stay, since `1+2*I` or `Rational(2/3)+I` should not evalf.

Reply:
Is this still an issue? You can have something like `1 + 2.0*I` where one part is Float and the other isn't.

Reply:
I had that concern because I didn't realize `m[0]` and `m[1]` are both at least `Number`. That's why I tweaked type hints of `pure_complex` afterwards.

Reply:
Oh I see. I didn't realize that `_prec` defaults to -1 in Number. I thought it was only defined on Float.